src/lib/dao.js

```diff
@@ -704,24 +699,20 @@ foam.CLASS({
 
       sink = this.decorateSink_(resultSink, options);
 
-      var promise = this.Promise.create();
-
       var fc = this.FlowControl.create();
       for ( var i = 0 ; i < this.array.length ; i++ ) {
         if ( fc.stopped ) break;
         if ( fc.errorEvt ) {
           sink.error(fc.errorEvt);
-          promise.reject(fc.errorEvt);
-          return promise;
+          return Promise.reject(fc.errorEvt);
         }
 
         sink.put(this.array[i], null, fc);
       }
 
-      sink.eof();
+      sink && sink.eof && sink.eof();
```

Comment:
Don't change this, let's force sinks to have eof for the time being.  I'm working on a plan that will allow uesrs to pass sinks with only a subset of the interface without having to do sink && sink.eof &&... every single time.